### PR TITLE
refactor: Migrate Metals server to Scala 3

### PR DESCRIPTION
**Pull request by tgodzik**
Just checking if it's possible and maybe let's start a conversation? It would be cool to migrate and show the world it's possible :D

Actually, this doesn't look that terrible.

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -20,10 +20,15 @@ project.excludeFilters = [
...
 fileOverride {
   "glob:**/sbt-metals/**" {
     runner.dialect = scala212source3
   },
+  "glob:**/scala-2*/**" {
+    runner.dialect = scala213
+  }
   "glob:**/scala-3*/**" {
     runner.dialect = scala3
     rewrite.scala3.convertToNewSyntax = yes
```
Comment (dos65) [inline .scalafmt.conf:34]: Move it default? :boom:

Comment (tgodzik) [reply .scalafmt.conf:34]: I was scared :P

**Comment by tgodzik** [comment]
Closing for now, let's wait for the improved warnings.

**Comment by kpodsiad** [comment]
@tgodzik what is eta for that?

**Comment by tgodzik** [comment]
> @tgodzik what is eta for that?

I keep asking that :sweat_smile:

**Pull request closed without merging**